utils/mock_qpu/infleqtion/__init__.py

```diff
@@ -0,0 +1,92 @@
+# ============================================================================ #
+# Copyright (c) 2022 - 2024 NVIDIA Corporation & Affiliates.                   #
+# All rights reserved.                                                         #
+#                                                                              #
+# This source code and the accompanying materials are made available under     #
+# the terms of the Apache License 2.0 which accompanies this distribution.     #
+# ============================================================================ #
+
+from fastapi import FastAPI, HTTPException, Header
+from typing import Union
+import uvicorn, uuid
+from pydantic import BaseModel
+
+# Define the REST Server App
+app = FastAPI()
+
+
+class Input(BaseModel):
+    format: str
+    data: str
+
+
+# Jobs look like the following type
+class Job(BaseModel):
+    qasm_strs: str
+    shots: int
+    target: str
+
+
+# Keep track of Job Ids to their Names
+createdJobs = {}
+
+# Could how many times the client has requested the Job
+countJobGetRequests = 0
+
+# Save how many qubits were needed for each test (emulates real backend)
+numQubitsRequired = 0
+
+
+# Here we test that the login endpoint works
+@app.post("/login")
+async def login(token: Union[str, None] = Header(alias="Authorization",
+                                                 default=None)):
+    if token == None:
+        raise HTTPException(status_code(401), detail="Credentials not provided")
+    return {"id-token": "hello", "refresh-token": "refreshToken"}
+
+
+# Here we expose a way to post jobs,
+# Must have a Access Token
+# with entry_point tag
+@app.post("/v0.2.0/jobs")
+async def postJob(job: Job,
+                  token: Union[str, None] = Header(alias="Authorization",
+                                                   default=None)):
+    global createdJobs, shots, numQubitsRequired
+
+    if token == None:
+        raise HTTPException(status_code(401), detail="Credentials not provided")
+
+    print('Posting job with shots = ', job.shots)
+    newId = str(uuid.uuid4())
+    # Job "created", return the id
+    return {"job_ids": [newId]}
+
+
+# Retrieve the job, simulate having to wait by counting to 3
+# until we return the job results
+@app.get("/v0.2.0/job/{id}")
+async def getJob(id: str):
+    print("Getting Job")
+    global countJobGetRequests, createdJobs, numQubitsRequired
+
+    # Simulate asynchronous execution
+    if countJobGetRequests < 3:
+        countJobGetRequests += 1
+        return {"status": "running"}
+
+    countJobGetRequests = 0
+    res = {'status': 'Done', "samples": {
```

Comment:
This is hardcoded because I don't know how to run OpenQASM in Cudaq. The other mock servers[ assumes that the input is QIR](https://github.com/bharat-thotakura/cuda-quantum/blob/4e5a5138916adcdb97f41bc4b0c19e71160e3476/utils/mock_qpu/braket/__init__.py#L83) so we can't learn from them.

Reply:
@bmhowe23 : Can you help here?

Reply:
We cannot currently run OpenQASM in our CUDA-Q runtime. One could, in principle, write conversion logic to convert OpenQASM to QIR and then run the QIR through our CUDA-Q runtime.

Reply:
How is this issue handled in braket? I see that it takes qasm as well https://github.com/NVIDIA/cuda-quantum/blob/f433d74e8e23a65fe5199062b7fdd7527c218740/runtime/cudaq/platform/default/rest/helpers/braket/braket.yml#L22

Reply:
The [Python tests here](https://github.com/NVIDIA/cuda-quantum/blob/main/python/tests/backends/test_braket.py#L18) are run against Amazon Braket server, not the mock server. (Hence skipped by the CI.) Any test(s) that uses `emulate` flag run QIR.

Reply:
We'll do what braket does. Thanks.